### PR TITLE
Fix fgets() usage

### PR DESCRIPTION
**Pull request by pelijah**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

File: libr/cons/input.c [hunk 1]
```diff
@@ -367,9 +367,9 @@ R_API int r_cons_fgets(char *buf, int len, int argc, const char **argv) {
 	*buf = '\0';
 	if (color) {
 		const char *p = cons->context->pal.input;
-		int len = p? strlen (p): 0;
-		if (len > 0) {
-			fwrite (p, len, 1, stdout);
+		int p_len = p? strlen (p): 0;
+		if (p_len > 0) {
```
Comment (ret2libc) [inline libr/cons/input.c]: use `R_STR_ISNOTEMPTY` instead of computing the len

File: libr/core/visual.c [hunk 2]
```diff
@@ -2548,7 +2548,7 @@ R_API int r_core_visual_cmd(RCore *core, const char *arg) {
 			r_line_set_prompt ("cmd.vprompt> ");
 			I->line->contents = strdup (cmd);
 			buf = r_line_readline ();
-//		if (r_cons_fgets (buf, sizeof (buf)-4, 0, NULL) <0) buf[0]='\0';
+//		if (r_cons_fgets (buf, sizeof (buf), 0, NULL) <0) buf[0]='\0';
```
Comment (ret2libc) [inline libr/core/visual.c]: please just delete

File: libr/core/visual.c [hunk 3]
```diff
@@ -2569,7 +2569,7 @@ R_API int r_core_visual_cmd(RCore *core, const char *arg) {
 				core->print->cur = 0;
 				(void)r_config_set (core->config, "cmd.cprompt", "p=e $r-2");
 			} else {
-				//		if (r_cons_fgets (buf, sizeof (buf)-4, 0, NULL) <0) buf[0]='\0';
+				//		if (r_cons_fgets (buf, sizeof (buf), 0, NULL) <0) buf[0]='\0';
```
Comment (ret2libc) [inline libr/core/visual.c]: delete

**Comment by radare** [comment]
looks green now

**Pull request merged**